test-d/partial-deep.ts

```diff
@@ -39,8 +39,8 @@ expectType<undefined>(partialDeepFoo.bar!.undefined);
 expectType<Map<string | undefined, string | undefined> | undefined>(partialDeepFoo.bar!.map);
 expectType<Set<string | undefined> | undefined>(partialDeepFoo.bar!.set);
 expectType<Array<string | undefined> | undefined>(partialDeepFoo.bar!.array);
-expectType<['foo' | undefined] | undefined>(partialDeepFoo.bar!.tuple);
+expectType<[('foo')?] | undefined>(partialDeepFoo.bar!.tuple);
```

Comment:
Any particular reason for parentheses here? You don't add them in the `readonly tuple` test (line 46).